addons/TrueFalse/src/presenter.js

```diff
@@ -1154,28 +1169,82 @@ function AddonTrueFalse_create() {
             var answers = [];
             if (showAnswers) answers = question.Answer.split(',');
 
-            for (var j = 0; j < model.Choices.length; j++) {
+            for (var j = 0; j < choiceLength; j++) {
                 var $td = $("<td></td>");
                 $td.addClass("checkbox-container");
                 $td.addClass("checkbox-" + (i+1) + "-" + (j+1));
                 var $inputDiv = $("<div></div>");
                 $inputDiv.addClass("placeholder");
                 $td.append($inputDiv);
                 var $checkbox = $("<input type=\"checkbox\"> </input>")
-                if (showAnswers && answers.indexOf((j+1).toString()) != -1) {
+                var userAnswerIndex = i * choiceLength + j;
+                if (didUserRespond && userAnswers[userAnswerIndex]) {
+                    $checkbox.attr("checked", "checked");
+                } else if (showAnswers && answers.indexOf((j+1).toString()) != -1 && !didUserRespond) {
                     $checkbox.attr("checked", "checked");
                 }
+
                 $td.append($checkbox);
                 var $checkboxSpan = $("<span></span>");
                 $td.append($checkboxSpan);
                 $tr.append($td);
+
+                if (showAnswers && isMulti && userAnswers[userAnswerIndex]) {
+                    var $markCell = $("<td></td>");
+                    var $markDiv = $("<div></div>");
+                    isAnswerCorrect(answers, userAnswers, i, j, choiceLength) ? $markDiv.addClass("correctAnswerMark") : $markDiv.addClass("incorrectAnswerMark");
+                    $markCell.append($markDiv);
+                    $tr.append($markCell);
+                } else if (showAnswers && didUserRespond && isMulti) {
+                    addCell(answers, userAnswers, $tr, i, choiceLength);
+                }
+            }
+            if (showAnswers && !isMulti && didUserRespondOnQuestion(userAnswers, i, choiceLength)) {
+                addCell(answers, userAnswers, $tr, i, choiceLength, true);
             }
             $tbody.append($tr);
         }
 
         $table.append($tbody);
         $view.append($table);
         return $view[0].outerHTML;
+
```

Comment:
1) daj te funkcje poza getPrintableHTML
2) teraz to dopiero zauważyłem, ale dobrze by było wyzerować presenter.printableState na samym końcu. Aktualnie jeżeli puści się drukowanie najpierw ze state'em, a potem bez, to za drugim razem nadal będzie korzystał ze state'a.
Reszta wygląda ok